server/src/main/java/org/elasticsearch/plugins/internal/DocumentParsingReporter.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugins.internal;
+
+/**
+ * An interface to allow performing an action when parsing has been completed and successful
+ */
+public interface DocumentParsingReporter {
```

Comment:
Since this only handles document size, can we make the name parallel to the observer as DocumentSizeReporter?